src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs

```diff
@@ -65,7 +66,14 @@ public override void VisitGenericName(GenericNameSyntax node)
 
         public override void VisitIdentifierName(IdentifierNameSyntax node)
         {
-            if (node.IsKind(SyntaxKind.IdentifierName) && TrySimplify(node))
+            // Always try to simplify identifiers with an 'Attribute' suffix.
+            //
+            // In other cases, don't bother looking at the right side of A.B or A::B. We will process those in
+            // one of our other top level Visit methods (like VisitQualifiedName).
+            var canTrySimplify = (node.Identifier.ValueText?.EndsWith("Attribute") ?? false)
```

Comment:
ValueText guaranteed to be non-null for an IdentifierToken.  it's an invariant of ROslyn (but can't be expressed with NRT).  Feel free to `!` away.